src/gp_doctest.cpp

```diff
@@ -1133,3 +1169,209 @@ TEST_CASE("NNI Engine: GraftDAG") {
+            if (nni_add >= nni_add_limit) {
+              break;
+            }
+            if (nni_add % test_after_every == 0) {
+              node_reindexer_without_root =
+                  node_reindexer.RemoveNewIndex(dag.GetDAGRootNodeId());
+              if (perform_reindex) {
+                gpengine.GrowPLVs(dag.NodeCountWithoutDAGRoot(),
+                                  node_reindexer_without_root);
+                gpengine.GrowGPCSPs(dag.EdgeCountWithLeafSubsplits(), edge_reindexer);
+              } else {
+                gpengine.GrowPLVs(dag.NodeCountWithoutDAGRoot());
+                gpengine.GrowGPCSPs(dag.EdgeCountWithLeafSubsplits());
+              }
+              // Test resizing and reindexing.
+              auto [post_node_map, post_edge_map, post_plv_map, post_branch_map] =
+                  BuildMapsFromDAG(dag, gpengine);
+              bool test_passes = CheckGPEngineResizeAndReindex(
+                  dag, gpengine, node_reindexer, edge_reindexer, pre_node_map,
+                  post_node_map, pre_edge_map, post_edge_map, pre_plv_map, post_plv_map,
+                  pre_branch_map, post_branch_map);
+              test_array.push_back(test_passes);
+              // Reinitialize reindexers.
+              node_reindexer = Reindexer::IdentityReindexer(dag.NodeCount());
+              edge_reindexer =
+                  Reindexer::IdentityReindexer(dag.EdgeCountWithLeafSubsplits());
+            }
+          }
+          nni_engine.ClearAdjacentNNIs();
+          nni_engine.SyncAdjacentNNIsWithDAG();
+          nni_count = nni_engine.GetAdjacentNNICount();
+
+          if (nni_add >= nni_add_limit) {
+            break;
+          }
+        }
+        // Test final resizing and reindexing.
+        node_reindexer_without_root =
+            node_reindexer.RemoveNewIndex(dag.GetDAGRootNodeId());
+        if (perform_reindex) {
+          gpengine.GrowPLVs(dag.NodeCountWithoutDAGRoot(), node_reindexer_without_root);
+          gpengine.GrowGPCSPs(dag.EdgeCountWithLeafSubsplits(), edge_reindexer);
+        } else {
+          gpengine.GrowPLVs(dag.NodeCountWithoutDAGRoot());
+          gpengine.GrowGPCSPs(dag.EdgeCountWithLeafSubsplits());
+        }
+        auto [post_node_map, post_edge_map, post_plv_map, post_branch_map] =
+            BuildMapsFromDAG(dag, gpengine);
+        bool test_passes = CheckGPEngineResizeAndReindex(
+            dag, gpengine, node_reindexer, edge_reindexer, pre_node_map, post_node_map,
+            pre_edge_map, post_edge_map, pre_plv_map, post_plv_map, pre_branch_map,
+            post_branch_map);
+        test_array.push_back(test_passes);
+        // Finally, test run full GP Optimization after all modifications completed.
+        inst.ReinitializePriors();
+        dag.ReinitializeTidyVectors();
+        inst.EstimateBranchLengths(0.0001, 100, true);
+        inst.PopulatePLVs();
+        inst.ComputeLikelihoods();
+        inst.ComputeMarginalLikelihood();
```

Comment:
@matsen This is where GP is run at the end of the test.

Reply:
"" 
I guess I didn't say it, but it would be nice to make sure that the algorithm gives the same values before and after PLV expansion.
""

Reply:
Right. I'm asking a different question.

Can we trigger a PLV expansion manually without editing the DAG? If so, could we trigger that and rerun things, checking to make sure the results are the same?

Reply:
Added test, pointed to it in comment below.